README.md

```diff
@@ -1,39 +1,167 @@
+1. We also recommend [PyCharm](http://www.jetbrains.com/pycharm/) because
+  * it's powerful.
+  * it's widely used in the group, so you can get help easily.
+  * as students you can request a free copy of the full production version!
+
+# Step 1: Fork the repository you wish to contribute
+Forking means you keep the entire copy of certain github repository under your account. 
+The advantages of forking are:
+1. it doesn't affect the original repository, i.e. what was working remains working.
+1. now this repository is completely under your account, you can do whatever you want. `fork` can be done through github website:
+
+![alt text](https://github.com/chiahaoliu/group_git_workflow/blob/doc_gitworkflow/img/git_fork.png "")
+
+# Step 2: Clone the git repository to local machine
+1. Go to the repository you wish to work on (ideally under your fork). Click on the green **clone or download** button at the top-right corner:
+
+![alt text](https://github.com/chiahaoliu/group_git_workflow/blob/doc_gitworkflow/img/git_clone.png "")
+
+1. The link shows here is the identity to this repository. You can then copy it and paste to your terminal and do:
+```
+git clone <url just copied>
+```
+Now you have your local copy of your remote repository!
+
+# Step 3: Set remote and fetch
+1. Navigate to your the directory you just cloned.
+1. As suggested by the name, `remote` means the repository on `github`. With a ``remote`` setup, you are able to compare the difference, push edits to and pull changes with respect to remote(s).
+1. To add a remote:
+```
+git remote add <name of this remote> <url to the github repository>
+```
+`url to the github repository` can be obtained as we did in the ``clone`` section.
+and ``name of this remote`` can be any name that is meaningful (easy to remember) to you.
+1. To view which remotes your local repository is synced to: 
+```
+git remote -a
+```
+And you might see something similar to this
+
+![alt text](https://github.com/chiahaoliu/group_git_workflow/blob/doc_gitworkflow/img/git_remote_v.png "")
+
+1. By default `origin` is always set to the repository you cloned. You can rename any `remote` by doing:
+```
+git remote rename <old name> <new name>
+```
+1. By convention, we usually set the parent repository of your fork as
+   `upstream`. Because that's where our contribution will go into.
+1. We then sync our local with all remotes. 
+```
+git remote update #update all remotes
+git fetch <remote name> #update certain remote
+```
+
+# Step4: Branch
+Branches are like virtual environments. Each of them has its own working history that is
+synced with certain `remote`. Once you ``checkout`` (switch) your branch, you
+are literally jumping into another parallel world where files live in *another* history.
+1. To list all available branches (including ones from all remotes):
+```
+git branch -a
+```
+1. To checkout (switch) to any branch available branch:
+```
+git checkout <branch name>
```

Comment:
@chiahaoliu The added steps from my last comment could go in as a note here instead I think